api/stripe.go

```diff
@@ -260,6 +260,8 @@ func (api *API) processStripeSub(ctx context.Context, s *stripe.Subscription) er
 	if err == nil {
 		api.staticLogger.Tracef("Subscribed user id %s, tier %d, until %s.", u.ID, u.Tier, u.SubscribedUntil.String())
 	}
+	// Re-set the tier cache for this user, in case their tier changed.
+	api.staticUserTierCache.Set(u)
```

Comment:
we should have an integration test for this. To make sure if you used up your tier but the next month starts, you are back to your paid tier instead of the anonymous one.